utilities/src/main/java/io/syndesis/qe/bdd/validation/DbValidationSteps.java

```diff
@@ -68,15 +71,96 @@ public void validateLead() {
 		Assertions.assertThat(getLeadTaskFromDb().toLowerCase()).isNotEmpty();
 	}
 
+	@Then("^validate add_lead procedure with last_name: \"([^\"]*)\", company: \"([^\"]*)\", period in ms: \"(\\w+)\"")
+	public void validateAddLeadProcedure(String lastName, String company, Integer ms) throws InterruptedException {
+		//wait for period cycle:
+		Thread.sleep(ms + 1000);
+		final long start = System.currentTimeMillis();
+		// We wait for at least 1 record to appear in DB (procedure goes on every 5 seconds).
+		final boolean contactCreated = TestUtils.waitForEvent(leadCount -> leadCount >= 1, () -> dbUtils.getNumberOfRecordsInTable(RestConstants.getInstance().getTODO_APP_NAME()),
+				TimeUnit.MINUTES,
+				2,
+				TimeUnit.SECONDS,
+				5);
+		Assertions.assertThat(contactCreated).as("Lead record has appeard in DB, todo table").isEqualTo(true);
+		Assertions.assertThat(getLeadTaskFromDb(lastName).contains(company));
+	}
+
+	@Then("^inserts into \"([^\"]*)\" table$")
+	public void insertsIntoTable(String tableName, DataTable data) throws SQLException {
+
+		List<List<String>> dataTable = data.raw();
+
+			String sql = null;
+			switch (tableName.toLowerCase()) {
+				case "todo":
+//				INSERT INTO TODOx(task) VALUES('Joe');
+					sql = "INSERT INTO todo(task) VALUES('%s'";
+					break;
+				case "contact":
+//				INSERT INTO CONTACT(first_name, last_name, company, lead_source) VALUES('Josef','Stieranka','Istrochem','db');
+					sql = "INSERT INTO CONTACT(first_name, last_name, company, lead_source) VALUES('%s'";
+					break;
+			}
+
+		Iterator it;
+		String next;
+		for (List<String> list:dataTable) {
+			it = list.iterator();
+			while (it.hasNext()) {
+				next = (String) it.next();
+				if (it.hasNext()) {
+					sql = String.format(sql, next) + ", '%s'";
+				} else {
+					sql = String.format(sql, next) + ")";
+				}
+			}
+		}
+		log.info("SQL query: *{}*", sql);
+		int newId = dbUtils.updateSqlOnSampleDb(sql);
+
+		//assert new row in database has been created:
+		Assertions.assertThat(newId).isEqualTo(1);
+	}
+
+	@Then("^validate that all todos with task \"([^\"]*)\" have value completed \"(\\w+)\", period in ms: \"(\\w+)\"$")
+	public void checksThatAllTodosHaveCompletedVal(String task, Integer val, Integer ms) throws InterruptedException {
+		Thread.sleep(ms + 1000);
+
+		ResultSet rs;
+		List<Integer> completedAll = new ArrayList<>();
+		try {
+			String sql = String.format("SELECT completed FROM todo WHERE task like %s", task);
+			rs = dbUtils.readSqlOnSampleDb(sql);
+			while (rs.next()) {
```

Comment:
I think this whole check can be simplified to 

```
Assertions.assertThat(rs).isEqualToComparingOnlyGivenFields(val, "completed");
```

Reply:
ok